### PR TITLE
Add set command to allow non-secure nodes to be controlled.

### PR DESCRIPTION
**Pull request by dhylands**
r? @julienw

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -210,6 +210,27 @@ fn main() {
                 }
             },
             "nodes_dbg"             => println!("{:?}\n", program.state.lock().unwrap().nodes),
+            "set"                   => {
+                if tokens.len() != 4 {
+                    println!("Must provide a HomeID, ValueID and a value");
```
Comment (julienw) [inline src/main.rs]: nit: use this as error message: `Syntax: set <home_id> <value_id> <value>`

File: src/main.rs [hunk 2]
```diff
@@ -210,6 +210,27 @@ fn main() {
...
+                        if let Ok(id) = u64::from_str_radix(tokens[2], 16) {
+                            let vid = ValueID::from_packed_id(home_id, id);
+                            let ref value_ids = program.state.lock().unwrap().value_ids;
+                            if value_ids.contains(&vid) {
+                                vid.set_string(tokens[3]);
+                            } else {
+                                println!("Unknown ValueID: {:08x} {:016x}", home_id, id);
+                            }
+                        } else {
+                            println!("Must specify a numberic value id");
```
Comment (julienw) [inline src/main.rs]: nit: "numeric"

File: src/main.rs [hunk 3]
```diff
@@ -210,6 +210,27 @@ fn main() {
                 }
             },
             "nodes_dbg"             => println!("{:?}\n", program.state.lock().unwrap().nodes),
+            "set"                   => {
+                if tokens.len() != 4 {
+                    println!("Must provide a HomeID, ValueID and a value");
+                } else {
+                    if let Ok(home_id) = u32::from_str_radix(tokens[1], 16) {
+                        if let Ok(id) = u64::from_str_radix(tokens[2], 16) {
```
Comment (julienw) [inline src/main.rs]: What do you think of this rewriting ? Trying to have less indentation:

``` rust
                if tokens.len() != 4 {
                    println!("Syntax: set <home_id> <value_id> <value>");
                    break;
                }

                match (u32::from_str_radix(tokens[1], 16), u64::from_str_radix(tokens[2], 16)) {
                    (Ok(home_id), Ok(id)) => {
                        let vid = ValueID::from_packed_id(home_id, id);
                        let ref value_ids = program.state.lock().unwrap().value_ids;
                        if value_ids.contains(&vid) {
                            vid.set_string(tokens[3]);
                        } else {
                            println!("Unknown ValueID: {:08x} {:016x}", home_id, id);
                        }
                    },
                    _ => println!("Must specify numeric ids")
                }
```

Comment (dhylands) [inline src/main.rs]: Ahh - didn't know about break with match

**Comment by dhylands** [comment]
Addressed review comments.

**Comment by dhylands** [comment]
Reworked to match @julienw 's suggested technique above

**Comment by julienw** [comment]
r=me !

**Pull request merged**